### PR TITLE
Second milestone

### PR DESCRIPTION
**Pull request by lucaasdfg**
second milestone added logic of the game and board

### CODE REVIEW

**Review by bolah2009 — CHANGES_REQUESTED** [review]
Let's add linter configuration files and setup linter.

[OPTIONAL]
Remember that this PR is about `User interface` not the `Game Logic`. I would recommend we keep it more basic and not use class.

#### :recycle:  Make these changes and submit another code review request. :+1: 

#### Feel free to reach me on slack if you having issues implementing these features. :+1: 

Good luck with your next submission. :star2: 

Cheers
[Bola](https://bolabuari.com/)

###### PS: Note that comments tagged as `OPTIONAL` are not part of the minimum requirements to approve this PR

**Review by uche-inyama — CHANGES_REQUESTED** [review]
## Changes Required

### Hi Lucas! :wave: 

Like my colleague pointed out, the aim of this milestone is to implement what the user will interact with and show the flow of the game. Let's give you a nudge :slightly_smiling_face: 

``` Initial instructions for the players ```

- [x] Game ask players for their names

 **A loop is needed in this game (recommended is `while` but anything else will do)**

``` while game on ```
* [ ] Inside loop, for each player, all the actions are repeated for each move
     - game asks player to select from available moves
     -  game informs player if selected move is invalid
     -   game displays board after player move

* [ ] A loop needs to be terminated at some point (after a winning move or draw to be more specific). 
    -  game informs player if selected move is a winning move
    -   game informs player if selected move is a draw move
    -  game repeats all actions for next player's move


                       
```ruby
game_on = true

# initial instructions for player

while game_on
  # loop for each move
  
  if winner # and/or draw (the exact condition in this milestone is not important)
    game_on = false
  end
end
```


Remember, implementing the game logic is not required, you can use comments to show how your logic works, like shown above :point_up_2: . 

For now, Interacting with the game upto the point where the board first appears is sufficient for this milestone.

What is required is that you show an understanding of the flow of the game.

Update your main.rb file remove the game class you created, it should be in the lib folder; it would be implemented in milestone 3.

After implementing these changes submit another form for a code review.

If you need clarifications, do not hesitate to reach out on slack, @Inyama, Uchechukwu Henry

#### ``` Have a great day ``` :slightly_smiling_face: 

``` Uchechukwu Henry Inyama ```

You have conflicts resolve them. :point_down:

File: bin/main.rb [hunk 1]
```diff
@@ -1,51 +1,46 @@
...
+# use sleep to create time on user interface changes
+sleep 2.0
+
+#  here display empy board : board
+
+# keep on while if 'result' is different from victory or draw, exits when get some of them
+while result != 'victory' || result != 'draw'
+
+  puts 'Mark your choice...'
+  input_choice = gets.chomp.downcase
```
Comment (samgaco) [inline bin/main.rb:73]: This is not leading anywhere in the user interface, even thought it's a mock, it should work on it's surface:

```

Hello. Welcome to Tic Tac Toe Game.

 Tell me your name...
santiago

 Okey, santiago. Now tell me your opponent name...
pedro

 Okey, santiago & pedro. {Some explanation on how to play goes here...a}Lets play.... 
Mark your choice...
a

Mark your choice...
O

Mark your choice...
1

Mark your choice...
2

Mark your choice...
X

Mark your choice...
a)        

Mark your choice...
???   

Mark your choice...

```

**Review by samgaco — CHANGES_REQUESTED** [review]
Hi @lucaasdfg and @alexawesomecode :high_brightness: 

- [ ] The user interface should work when executed (get until the point that shows a winner or a draw, etc, even though no real logic is needed behind it)

- [ ] Please, solve the conflicts before submitting them back for another code review.

-------
* ###### STATUS OF THE PROJECT: REQUESTED CHANGES :stop_sign:

**Comment by alexawesomecode** [comment]
Hello @samgaco 

So we have to check for 'valid user inputs' and for 'checking for win or draw'? We have some methods working that do that work. So we could include those methods here. Will that meet the requirements?

Thanks for your attention

**Review by maelfosso — APPROVED** [review]
# Congratulations :clap: Is approved :smile: 

You can merge it now :+1: 
Good luck with the next milestone 

**PS**: For the next milestone, Don't use global variables, more think about OOP, Single Responsibility principle. All the logic should come from the classes.

**Mael FOSSO**
[Github](https://github.com) | [Twitter](https://twitter.com) | [LinkedIn](https://www.linkedin.com/in/mael-elvis-fosso-650b6346/)

**Pull request merged**